core/src/main/java/com/backblaze/b2/client/structures/B2Capability.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+
+package com.backblaze.b2.client.structures;
+
+import com.backblaze.b2.json.B2JsonException;
+import com.backblaze.b2.json.B2JsonReader;
+import com.backblaze.b2.json.B2JsonTypeHandler;
+import com.backblaze.b2.json.B2JsonWriter;
+import com.backblaze.b2.util.B2StringUtil;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public enum B2Capability {
```

Comment:
i'm a big fan of having strong typing when we can and enums are better for that than strings.  With that said...please read the "DESIGN NOTE" comment about compatibility in B2BucketTypes.java and then let's talk.  :)